lib/reline/line_editor.rb

```diff
@@ -1137,7 +1138,10 @@ def input_key(key)
       @completion_journey_state = nil
     end
 
-    push_past_lines unless @undoing
+    unless @undoing
+      @past_lines = @past_lines[0..@position]
```

Comment:
I think this part needs to be implemented inside `push_past_lines` because:
- `push_past_lines` is also called from `insert_pasted_text`. This part is also needed in there.
- This will blow redo history even when cursor is just moved. This part should be done only when `@old_buffer_of_lines != @buffer_of_lines`

Reply:
Thank you for your comment. I've fixed it.

[Fixed deletion of the redo history in regular input](https://github.com/ruby/reline/pull/707/commits/720cd98a36bdc012de2201466011074d1daa5f05)